### PR TITLE
support querying elements in an array

### PR DESCRIPTION
**Pull request by haraldschilly**
# Description

this is necessary for the CRM data, to filter the accounts/tags field

my main worry is, I hope this does not break anything else.

Screenshot from 2024-04-12 18-27-40

## [Checklist](https://github.com/sagemathinc/cocalc/wiki/PR-Checklist):
- [ ] Testing instructions are provided, if not obvious
- [ ] Release instructions are provided, if not obvious

### CODE REVIEW

**Comment by williamstein** [comment]
Yes, this will be nice!

**Comment by williamstein** [comment]
(The other nearby missing big feature is AND queries involving the same field twice.  I never got to that...)

**Comment by haraldschilly** [comment]
Yes, I see, but that's for another day. In any case, FWIW all tests pass. You can check for yourself, I tried to keep this patch minimal.

File: src/packages/frontend/frame-editors/crm-editor/views/view-menu/search.tsx [hunk 1]
```diff
@@ -151,12 +151,15 @@ function SearchBy({
...
+    if (fieldSpec.type === "boolean") {
       return [
         { value: "IS" as Operator, label: "IS" },
         { value: "IS NOT" as Operator, label: "IS NOT" },
       ];
     }
+    if (fieldSpec.type === "array") {
+      return [{ value: "ANY" as Operator, label: "ANY" }];
+    }
     return OPERATORS.filter((op) => op != "==").map((op: Operator) => {
```
Comment (haraldschilly) [inline src/packages/frontend/frame-editors/crm-editor/views/view-menu/search.tsx:163]: oh I see, here it appears again. So, maybe what's left is to filter "ANY" like "==". … but that shouldn't be a real problem...

**Pull request merged**